code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java

```diff
@@ -225,6 +236,35 @@ protected void modifyDisk(long hostId, Instance instance, boolean add) {
 
     }
 
+    @Inject
+    IdFormatter idFormatter;
+
+    public void addReleaseSchedulerResource(String action, Long instanceId, boolean isVM, Long hostId, Long envId) {
+        String ALLOCATE_CPU_MEMORY_URL = SCHEDULER_URL + "/" + action;
+        List<BasicNameValuePair> requestData = new ArrayList<>();
+
+        requestData.add(new BasicNameValuePair("hostId", (String) idFormatter.formatId(HostConstants.TYPE, hostId)));
+        requestData.add(new BasicNameValuePair(isVM ? "vmId" : "instanceId",
+                (String) idFormatter.formatId(InstanceConstants.TYPE, instanceId)));
+        requestData.add(new BasicNameValuePair("envId", (String) idFormatter.formatId(AccountConstants.TYPE, envId)));
+
+        HttpResponse response;
+        try {
+            response = Request.Post(ALLOCATE_CPU_MEMORY_URL).addHeader("Accept", "application/json")
+                    .bodyForm(requestData).execute().returnResponse();
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (statusCode != 200) {
+                log.error("statusCode: {}", statusCode);
+            }
+        } catch (HttpHostConnectException ex) {
+            log.error("Scheduler Service not reachable at [{}]", ALLOCATE_CPU_MEMORY_URL);
+        } catch (IOException e) {
+            log.error((e.getStackTrace()).toString(), e);
+        }
+
```

Comment:
Are you saying if statusCode != 200, we should throw exception ? Yes, if there is a http or I/O exception, we log them as well. I think this is best effort operation.